### ISSUE TITLE
Hiding the window makes it impossible to open it back up on Debian 12 on Wayland

### ISSUE THREAD

**Issue by colfrog** [spotlightify/spotlightify#113]
I believe that if an instance is already running, on startup Spotlightify (WebKit) tries to send a signal through the X window system. This does not work on Wayland, neither on GNOME nor on KDE.

This makes the program unusable on Wayland on Debian 12.

My current fix is to replace the calls to Hide in the commands to Quit, and move them after the await that sends the command to the Spotify API. This is an ugly fix, but when your hotkey invokes the Spotlightify program it works.

I don't think that my solution is desired for the project, it might not be the best for other operating systems and I think it might mess with the suggestions, but I'd like to start a conversation on how the program should be working on Wayland. Maybe WebKit has fixed this in later releases, needs testing.

**Comment by petergmurphy** [comment]
> I believe that if an instance is already running, on startup Spotlightify (WebKit) tries to send a signal through the X window system.

This is correct. Currently I'm using a [hotkey library](https://pkg.go.dev/golang.design/x/hotkey) which relies on X11 to listen for keybindings on Linux. I'm in the process of looking at Wails V3 (still in Alpha but close to Beta) to see if it's stable enough to migrate to, it has built in hotkey listening features which could solve these compatibility issues.
